src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -270,17 +270,27 @@ float64_t CCrossValidation::evaluate_one_run()
 			CMachine* machine;
 			CFeatures* features;
 			CLabels* labels;
+			CEvaluation* evaluation_criterion;
 
 			if (get_global_parallel()->get_num_threads()==1)
+			{
 				machine=m_machine;
+				features=m_features;
+				evaluation_criterion=m_evaluation_criterion;
+			}
 			else
+			{
 				machine=(CMachine*)m_machine->clone();
+				features=(CFeatures*)m_features->clone();
+				evaluation_criterion=(CEvaluation*)m_evaluation_criterion->clone();
```

Comment:
Does `clone` actually return a zero a one refcount?
If it is zero, it should be `SG_REF`ed here as well, same for features